gravityflow/gravityflow
# Fix role tokens

This PR fixes an issue which prevents tokens from working correctly with role assignees. It also refactors some variable names to clarify the context.

Fixes [HS6113](https://secure.helpscout.net/conversation/590941529/6113?folderId=1113498)

The acceptance test bootstrap now loads mu-plugins from the mu-plugins folder. This PR includes a plugin to create a post for each Gravity Forms notification which enables us to write tests for the tokens and email contents.

**Testing instructions**
To reproduce the issue on the master branch, create a form, add an approval step and assign it to a role. In the assignee email use the token attribute in an entry link merge tag: {workflow_entry_link: token="true"}. Submit the form and open the link in the email in a different browser while not logged in.
The assignee can view the entry but not approve because the buttons are missing.

On this branch, the buttons appear and the assignee can approve.

Check for regression issues with different step types, assignee types: user, multi-user & email fields combined with different assignee policies.
